src/modules/job-manager/submit.c

```diff
@@ -207,11 +261,15 @@ static void submit_cb (flux_t *h, flux_msg_handler_t *mh,
         flux_log_error (h, "%s: error creating newjobs list", __FUNCTION__);
         goto error;
     }
+    if (submit_validate_jobs (ctx, newjobs, &errors) < 0) {
+        flux_log_error (h, "%s: error validating batch", __FUNCTION__);
+        goto error;
+    }
     if (submit_hash_jobs (ctx->active_jobs, newjobs) < 0) {
         flux_log_error (h, "%s: error enqueuing batch", __FUNCTION__);
         goto error;
     }
-    if (flux_respond (h, msg, "{}") < 0)
+    if (flux_respond_pack (h, msg, "{s:o}", "errors", errors) < 0)
```

Comment:
not a huge deal, but in `job-ingest` you have:

```
    /*  Empty payload indicates the whole batch was successful
     */
    if (o == NULL)
        return bresp;

    /*  O/w, there were one or more failures, capture these in the
     *   the response errors hash.
     */
```

it appears that you are always sending back an errors array, even if there are no errors?  Did you want to not send an array back when there are no errors?  Or perhaps just tweak the comment in `job-ingest`?

Reply:
The dummy job-manager module used in testing sends back an empty payload.

Reply:
Ok, so I guess the only thing would be the comment "O/w there were one or more failures" in job-ingest.  B/c you may be sending back zero errors in the array?

Reply:
Got it. Good point, I didn't actually re-read the comment :facepalm: